### PR TITLE
Add Initial Implementation of sage_sms Package for SMS Provider Integration

### PR DESCRIPTION
**Pull request by MohmdFo**
#### Summary
This pull request introduces the initial implementation of the `sage_sms` package, designed to facilitate the integration of different SMS providers. The package provides a structured and modular approach to managing SMS providers, ensuring easy extension and maintenance.

#### Impact
This initial implementation lays the groundwork for the `sage_sms` package, providing a clear and organized structure. Future contributions can build upon this foundation by adding specific SMS provider integrations, improving the utility and flexibility of the package.

#### Next Steps
- Add unit tests for core components.
- Enhance documentation.

Please review the changes and provide feedback. Your input is invaluable in refining this initial implementation and guiding future developments.

### CODE REVIEW

File: sage_sms/factory.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+from sage_sms.helper.exceptions import (
+    SMSBackendError,
+    SMSConfigurationError,
+    SMSProviderNotFoundError,
+    SMSUnexpectedError,
+)
+from sage_sms.helper.type import ProviderSettings, SMSSettings
+
+# Configure logging
+logging.basicConfig(level=logging.DEBUG)
```
Comment (sepehr-akbarzadeh) [inline sage_sms/factory.py]: remove this line!

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,89 @@
+[tool.poetry]
+name = "python-sage-sms"
+version = "0.1.0"
+description = "A Python package for sending SMS with different providers"
+authors = [
+    "Sepehr Akbarzadeh <sa.goldeneagle@gmail.com>",
```
Comment (sepehr-akbarzadeh) [inline pyproject.toml]: fix email addresses. it must be organizational emails for members!

File: mypy.ini
Comment (sepehr-akbarzadeh) [inline mypy.ini:1]: remove this file!

File: .pylintrc
Comment (sepehr-akbarzadeh) [inline .pylintrc:1]: remove this file!

File: .flake8
Comment (sepehr-akbarzadeh) [inline .flake8:1]: remove this file!

File: setup.cfg
Comment (sepehr-akbarzadeh) [inline setup.cfg:1]: remove this file!

File: setup.py
Comment (sepehr-akbarzadeh) [inline setup.py:1]: remove this file

**Review by sepehr-akbarzadeh — CHANGES_REQUESTED** [review]

**Comment by sepehr-akbarzadeh** [comment]
Thank you for your collaboration on this project.

I have reviewed your PR and **found 7 issues** that need to be addressed. Could you please fix these issues first and then provide the updated PR?

Thanks again for your hard work and dedication.

**Comment by MohmdFo** [comment]
I have completed all the requests mentioned above. I'd appreciate it if you could review them again.
@sepehr-akbarzadeh

**Pull request merged**